scripts/inject-deterministic.py

```diff
@@ -4,14 +4,129 @@
+
+        return script_with_nouce
+
+    def get_nonce_from_headers(self, header):
+        """
+        get_nonce_from_headers returns the nonce token from a
+        Content-Security-Policy (CSP) header's script source directive, or an empty
+        string if the CSP header's script source does not contain a nonce.
+        For more background information on CSP and nonce, please refer to
+        https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        Content-Security-Policy/script-src
+        https://developers.google.com/web/fundamentals/security/csp/
+        """
+
+        csp_script_directives = self.get_csp_script_directives(header)
+        if not csp_script_directives == "":
+            tokens = csp_script_directives.split(" ")
+            for token in tokens:
+                if token.startswith("'nonce-"):
+                    token = token.lstrip("'nonce-")
+                    token = token.rstrip("'")
+                    return token
+        return ""
+
+    def get_csp_script_directives(self, headers):
+        # get CSP script-src directives
+        if "Content-Security-Policy" in headers:
+            ctx.log.info("Original Header %s" % headers["Content-Security-Policy"])
+            csp = headers["Content-Security-Policy"]
+            directives = csp.split(";")
+
+            for directive in directives:
+                if directive.startswith("script-src "):
+                    return directive
+
+        return ""
+
+    def get_csp_header(self, headers, sha256):
+        # get_csp_header transforms a Content-Security-Policy (CSP) header to
+        # permit execution of inline scripts. Without this permission a page with a
+        # restrictive CSP will not execute injected scripts.
+        # For more background information on CSP, please refer to
+        # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        # Content-Security-Policy/script-src
+        # https://developers.google.com/web/fundamentals/security/csp/
+
+        header_updated = False
+        if "Content-Security-Policy" in headers:
+            csp = headers["Content-Security-Policy"]
+            csp_directives = csp.split(";")
+        else:
+            return headers
+
+        for id, directive in enumerate(csp_directives):
+            if directive.startswith("script-src "):
+                tokens = directive.split(" ")
+                new_directive = ""
+                add_unsafe = True
+
+                for token in tokens:
+                    token = token.strip()
+                    if token == "'unsafe-inline'":
+                        add_unsafe = False
+                        ctx.log.info("Has unsafe-inline")
+
+                    if token.startswith("'sha256-") or \
+                            token.startswith("'sha384-") or \
+                            token.startswith("'sha512-"):
+                        new_directive += "'sha256-" + sha256 + "' "
```

Comment:
We should be able to modify the sources in place, and then join them to create the new directive.